kaldb/src/main/java/com/slack/kaldb/bulkIngestApi/opensearch/BulkApiRequestParser.java

```diff
@@ -103,6 +104,27 @@ public static Trace.Span fromIngestDocument(
     spanBuilder.setTimestamp(
         TimeUnit.MICROSECONDS.convert(timestampInMillis, TimeUnit.MILLISECONDS));
 
+    if (sourceAndMetadata.get(LogMessage.ReservedField.PARENT_ID.fieldName) != null) {
+      spanBuilder.setParentId(
+          ByteString.copyFromUtf8(
+              (String) sourceAndMetadata.get(LogMessage.ReservedField.PARENT_ID.fieldName)));
+      sourceAndMetadata.remove(LogMessage.ReservedField.PARENT_ID.fieldName);
+    }
+    if (sourceAndMetadata.get(LogMessage.ReservedField.TRACE_ID.fieldName) != null) {
+      spanBuilder.setTraceId(
+          ByteString.copyFromUtf8(
+              (String) sourceAndMetadata.get(LogMessage.ReservedField.TRACE_ID.fieldName)));
+      sourceAndMetadata.remove(LogMessage.ReservedField.TRACE_ID.fieldName);
+    }
+    if (sourceAndMetadata.get(LogMessage.ReservedField.NAME.fieldName) != null) {
+      spanBuilder.setName((String) sourceAndMetadata.get(LogMessage.ReservedField.NAME.fieldName));
+      sourceAndMetadata.remove(LogMessage.ReservedField.NAME.fieldName);
+    }
+    if (sourceAndMetadata.get("duration") != null) {
+      spanBuilder.setDuration(Long.parseLong(sourceAndMetadata.get("duration").toString()));
+      sourceAndMetadata.remove("duration");
```

Comment:
I'm not sure how I feel about removing these from the tags - what if we leave these in the tags as well?

Reply:
if these get left as tags, eventually when we create a lucene document in SchemaAwareLogDocumentBuilderImpl#fromMessage , we need to make sure those fields don't get added twice. 

I personally would leave this as is for now. It's really brittle though. Depending on mileage this month we could revisit BulkApiRequestParser#fromIngestDocument and  SchemaAwareLogDocumentBuilderImpl#fromMessage to work more in harmony. Part of that will be evolving trace.proto to be a more flat structure